fuzz.go

```diff
@@ -190,6 +193,15 @@ func (f *Fuzzer) SkipFieldsWithPattern(pattern *regexp.Regexp) *Fuzzer {
 	return f
 }
 
+// WithCharRange is used to set the Unicode encoding range of the string.
+func (f *Fuzzer) WithCharRange(first, last rune) *Fuzzer {
```

Comment:
Here's another option to think about, what if we added a helper function to construct a string randomization function?

```go

type UnicodeRange [2]rune // (could also be like the existing charRange type)
type UnicodeRanges []UnicodeRange

// CustomStringFuzzFunc constructs a function which randomizes strings.
// Example usage: `cjk := UnicodeRanges{{0x4e00, 0x9fff}}; f.Funcs(cjk.CustomStringFuzzFunc())
func (r UnicodeRanges) CustomStringFuzzFunc() func(s *string, c fuzz.Continue) {
  return func(s *string, c fuzz.Continue) {
    ...
  }
}
```

This would replace the existing charRange type.

I think I'd rather make it easy for people to customize instead of adding more features to the fuzzer.

Reply:
Well, it is good idea.

I delete function `WithCharRange`, add structure `UnicodeRange` and slice `UnicodeRanges`.Each of them has function `CustomStringFuzzFunc`

If people who just wants to generate a string with one encoding, they can use `UnicodeRange` function `CustomStringFuzzFunc` to generate. 
Those who want to generate a string in different encoding, they can use `UnicodeRanges` function `CustomStringFuzzFunc` to generate. 
If people don't want to custom string, they can use `f.Fuzz(myString)` as usual.

The changes have push, please @lavalamp review and comment !